src/transforms/uglify.js

```diff
@@ -11,18 +13,39 @@ module.exports = async function(asset) {
     warnings: true,
     mangle: {
       toplevel: true
-    }
+    },
+    sourceMap: asset.options.sourceMaps ? {filename: asset.relativeName} : false
   };
 
   if (customConfig) {
     options = Object.assign(options, customConfig);
   }
 
   let result = minify(code, options);
+
   if (result.error) {
     throw result.error;
   }
 
+  if (result.map) {
+    result.map = await new SourceMap().addMap(JSON.parse(result.map));
+    if (asset.sourceMap) {
+      asset.sourceMap = await new SourceMap().extendSourceMap(
+        asset.sourceMap,
+        result.map
+      );
+    } else {
+      asset.sourceMap = result.map;
+    }
+  }
+
+  // Log all warnings
```

Comment:
I'd rather not log all of these. Uglify logs a lot of warnings and they are pretty much useless. I removed this in master already.